backend/ethereum/channel/adjudicator.go

```diff
@@ -81,7 +83,11 @@ func (a *Adjudicator) Progress(ctx context.Context, req channel.ProgressReq) err
 		state adjudicator.ChannelState,
 		_ [][]byte,
 	) (*types.Transaction, error) {
-		return a.contract.Progress(opts, params, state, ethNewState, ethActorIndex, req.Sig)
+		sigBytes, err := req.Sig.MarshalBinary()
+		if err != nil {
+			return nil, errors.WithMessage(err, "converting to byte array")
+		}
+		return a.contract.Progress(opts, params, state, ethNewState, ethActorIndex, sigBytes)
```

Comment:
This looks a bit like you are mixing concerns here. MarshalBinary should typically be used by a marshaler that doesn't know much about the concrete type and just wants **some** byte representation. However, here we know the type and the contract expects a very specific byte representation.
```suggestion
		sig, ok := req.Sig.(*ethwallet.Sig)
		if !ok {
			return nil, errors.Errorf("invalid signature type: %T", req.Sig)
		}
		return a.contract.Progress(opts, params, state, ethNewState, ethActorIndex, *sig)
```